KrispyTech/airneis
# AIRNEIS-127 - feat(back - auth): Sign-In

# DONE

## Features
- Logger for all API requests
- Handler for external services support to use in routes
- Add new custom lib to use for error handling 
- Added Hashicorp Vault integration to fetch secrets needed for JWT authentication

JWT: 
- Have a serialized user for JWT returns
- Added validation to check if JWT has a correct signature, also checks expiration for JWT. 
- Authentication on our `/api/session` endpoint.
  - Use POST request to `/api/session` with an email and password body parameter to login
  - Use DELETE request to `/api/session` to logout


## Reworks
- Fiber app initialization, made changes to handle better maintainability and dev experience
- Rework for better error handling
- Rework on controllers, separate each package. 
- Assemble routes and adapt them to the new controllers rework

# QA

- Have a user created on the database and make a POST request to `/api/session` with an email and password body parameter to try to login.
- Check if when you make a HTTP DELETE request to `/api/session` to logout, the session cookie is removed.
